--- a/src/oauth2.erl
+++ b/src/oauth2.erl
@@ -135,7 +135,7 @@
     case auth_client(Client, no_redir, Ctx0) of
         {error, _}      -> {error, invalid_client};
         {ok, {Ctx1, C}} ->
-            case ?BACKEND:verify_client_scope(C, Scope0, Ctx1) of
+            case ?BACKEND:verify_client_scope(Client, Scope0, Ctx1) of
                 {error, _}           -> {error, invalid_scope};
                 {ok, {Ctx2, Scope1}} ->
                     {ok, {Ctx2, #a{ client=C
--- a/test/oauth2_mock_backend.erl
+++ b/test/oauth2_mock_backend.erl
@@ -118,8 +118,8 @@
 verify_redirection_uri({client, 4711}, ?CLIENT_URI, Ctx) -> {ok, Ctx};
 verify_redirection_uri(_, _, _)                          -> {error, mismatch}.
 
-verify_client_scope({client, 4711}, [], Ctx)      -> {ok, {Ctx, []}};
-verify_client_scope({client, 4711}, ?CSCOPE, Ctx) -> {ok, {Ctx, ?CSCOPE}};
+verify_client_scope({?CID, ?SECRET}, [], Ctx)      -> {ok, {Ctx, []}};
+verify_client_scope({?CID, ?SECRET}, ?CSCOPE, Ctx) -> {ok, {Ctx, ?CSCOPE}};
 verify_client_scope(_, _, _)                       -> {error, invalid_scope}.
 
 verify_resowner_scope({user, 31337}, ?USCOPE, Ctx) -> {ok, {Ctx, ?USCOPE}};
